nodes/analyzer/weighted_vector_sum.py

```diff
@@ -0,0 +1,453 @@
+    def draw_buttons(self, context, layout):
+        col = layout.column()
+
+        row = col.row()
+        row.prop(self, 'skip_unmanifold_centers')
+
+        row = col.row()
+        split = row.split(factor=0.4)
+        c1 = split.column()
+        c1.label(text="Center mode:")
+        c2 = split.column().row(align=True)
+        c2.prop(self, "center_mode", text='')
+
+        row = col.row()
+        row.active = False
+        row.label(text='Triangulation (only Faces or Volumes):')
+        if self.center_mode in ['FACES', 'VOLUMES']:
+            row.active = True
+
+        row = col.row()
+        row.active = False
+        split = row.split(factor=0.4)
+        c1 = split.column()
+        c1.label(text="Quads mode:")
+        c2 = split.column().row(align=True)
+        c2.prop(self, "quad_mode", text='')
+        if self.center_mode in ['FACES', 'VOLUMES']:
+            row.active = True
+
+        row = col.row()
+        row.active = False
+        split = row.split(factor=0.5)
+        c1 = split.column()
+        c1.label(text="Polygons mode:")
+        c2 = split.column().row(align=True)
+        c2.prop(self, "ngon_mode", text='')
+        if self.center_mode in ['FACES', 'VOLUMES']:
+            row.active = True
+
+        # layout.prop(self, "quad_mode")
+        # layout.prop(self, "ngon_mode")
+        pass
+
+    def sv_init(self, context):
+        self.width=220
+        self.inputs.new('SvVerticesSocket', 'input_vertices')
+        self.inputs.new('SvStringsSocket', 'input_edges')
+        self.inputs.new('SvStringsSocket', 'input_polygons')
+        self.inputs.new('SvStringsSocket', 'input_mass_vert')
+        self.inputs.new('SvStringsSocket', 'input_density')
+
+        self.inputs["input_vertices"] .label = 'Vertices'
+        self.inputs["input_edges"]    .label = 'Edges'
+        self.inputs["input_polygons"] .label = 'Polygons'
+        self.inputs["input_mass_vert"].label = 'Mass of vertiсes'
+        self.inputs["input_density"]  .label = 'Density'
+
+
+        self.outputs.new('SvVerticesSocket', 'output_vertices')
+        self.outputs.new('SvStringsSocket',  'output_edges')
+        self.outputs.new('SvStringsSocket',  'output_polygons')
+
+        self.outputs.new('SvVerticesSocket', 'output_centers_of_mass')
+        self.outputs.new('SvVerticesSocket', 'output_total_center')
+        self.outputs.new('SvStringsSocket', 'output_sizes')
+        self.outputs.new('SvStringsSocket', 'output_size')
+        self.outputs.new('SvStringsSocket', 'output_masses')
+        self.outputs.new('SvStringsSocket', 'output_mass')
+        self.outputs.new('SvStringsSocket', 'output_mask')
+
+        self.outputs['output_vertices'] .label = 'Vertices'
+        self.outputs['output_edges'] .label = 'Edges'
+        self.outputs['output_polygons'] .label = 'Polygons'
+        self.outputs['output_centers_of_mass'].label = 'Centers mass of objects'
+        self.outputs['output_total_center'].label = 'Total center'
```

Comment:
"Common center"?

Reply:
So many centers. I mean “Center of all objects”. May be last is good?

Reply:
Ok.

Reply:
Is it good?
![image](https://github.com/nortikin/sverchok/assets/14288520/7d5f7d45-e722-45b4-b86a-41fd87718ffd)